# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [4]:
# "tobs" is "temperature observations"
df = pd.read_csv("C:/Users/krcon/OneDrive/Desktop/Class Master Folder/Sqlalchemy-Challenge/Resources/hawaii_measurements.csv")
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [5]:
# Convert the date column format from string to datetime
df["date"]=pd.to_datetime(df["date"])

In [6]:
# Set the date column as the DataFrame index
df= df.set_index(df["date"])
df.head()

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

In [7]:
# Drop the date column
df= df.drop(columns='date')
df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [10]:
from scipy import stats


In [25]:
# Filter data for desired months
june=df[df.index.month==6]
june.head(30)

station  prcp  tobs
date                               
2010-06-01  USC00519397  0.00    78
2010-06-02  USC00519397  0.01    76
2010-06-03  USC00519397  0.00    78
2010-06-04  USC00519397  0.00    76
2010-06-05  USC00519397  0.00    77
2010-06-06  USC00519397  0.00    78
2010-06-07  USC00519397  0.00    77
2010-06-08  USC00519397  0.00    78
2010-06-09  USC00519397  0.00    78
2010-06-10  USC00519397  0.00    79
2010-06-11  USC00519397  0.00    78
2010-06-12  USC00519397  0.00    78
2010-06-13  USC00519397  0.00    78
2010-06-14  USC00519397  0.00    77
2010-06-15  USC00519397  0.00    78
2010-06-16  USC00519397  0.00    78
2010-06-17  USC00519397  0.00    77
2010-06-18  USC00519397  0.00    77
2010-06-19  USC00519397  0.00    82
2010-06-20  USC00519397  0.00    78
2010-06-21  USC00519397  0.00    78
2010-06-22  USC00519397  0.00    78
2010-06-23  USC00519397  0.00    78
2010-06-24  USC00519397  0.00    78
2010-06-25  USC00519397  0.00    77
2010-06-26  USC00519397  0.02    76
2010-06-27  USC00519397  0.00    78
2010-06-28  USC00519397  0.00    78
2010-06-29  USC00519397  0.00    78
2010-06-30  USC00519397  0.00    78

In [26]:
december= df[df.index.month==12]
december.head(30)

station  prcp  tobs
date                               
2010-12-01  USC00519397  0.04    76
2010-12-03  USC00519397  0.00    74
2010-12-04  USC00519397  0.00    74
2010-12-06  USC00519397  0.00    64
2010-12-07  USC00519397  0.00    64
2010-12-08  USC00519397  0.00    67
2010-12-09  USC00519397  0.00    77
2010-12-10  USC00519397  1.58    66
2010-12-11  USC00519397  0.77    69
2010-12-12  USC00519397  0.00    68
2010-12-13  USC00519397  0.00    68
2010-12-14  USC00519397  0.00    71
2010-12-15  USC00519397  0.00    74
2010-12-16  USC00519397  0.00    67
2010-12-17  USC00519397  0.00    66
2010-12-18  USC00519397  0.00    69
2010-12-19  USC00519397  1.40    71
2010-12-23  USC00519397  0.00    70
2010-12-24  USC00519397  0.00    70
2010-12-26  USC00519397   NaN    74
2010-12-27  USC00519397  0.13    74
2010-12-28  USC00519397  0.55    71
2010-12-29  USC00519397  0.06    75
2010-12-30  USC00519397  0.01    75
2010-12-31  USC00519397  0.00    72
2011-12-01  USC00519397  0.03    69
2011-12-02  USC00519397  0.08    68
2011-12-03  USC00519397  0.00    73
2011-12-04  USC00519397  0.00    74
2011-12-05  USC00519397  0.00    73

In [16]:
# Identify the average temperature for June
juneavg= june['tobs'].mean()
print(juneavg)

74.94411764705882


In [17]:
# Identify the average temperature for December
decavg=december['tobs'].mean()
print(decavg)

71.04152933421226


In [22]:
# Create collections of temperature data
junet= june.tobs
junet

date
2010-06-01    78
2010-06-02    76
2010-06-03    78
2010-06-04    76
2010-06-05    77
              ..
2017-06-26    79
2017-06-27    74
2017-06-28    74
2017-06-29    76
2017-06-30    75
Name: tobs, Length: 1700, dtype: int64

In [23]:
dect=december.tobs
dect

date
2010-12-01    76
2010-12-03    74
2010-12-04    74
2010-12-06    64
2010-12-07    64
              ..
2016-12-27    71
2016-12-28    71
2016-12-29    69
2016-12-30    65
2016-12-31    65
Name: tobs, Length: 1517, dtype: int64

In [24]:
# Run paired t-test
stats.ttest_ind(dect,junet)

Ttest_indResult(statistic=-31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

A 3.9 degrees difference in average temperature in June than December indicates a slight but very little variance based on the time of year visiting Hawaii. 